environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java

```diff
@@ -621,10 +629,20 @@ public Weld addAlternativeStereotype(Class<? extends Annotation> alternativeSter
      * @see #ARCHIVE_ISOLATION_SYSTEM_PROPERTY
      * @see #SHUTDOWN_HOOK_SYSTEM_PROPERTY
      * @see #DEV_MODE_SYSTEM_PROPERTY
+     * @see #ADDITIONAL_BEAN_DEFINING_ANNOTATIONS_PROPERTY
      * @see ConfigurationKey
      */
     public Weld property(String key, Object value) {
         properties.put(key, value);
+        if (ADDITIONAL_BEAN_DEFINING_ANNOTATIONS_PROPERTY.equals(key) && value instanceof Collection) {
```

Comment:
No other properties are parsed here, so I'd stick with that and move this logic over to the `initialize()` method.
You're gonna have to reshuffle it there a bit (or I can if you don't mind me adding to this PR) - parse properties sooner then `createDeployment()` is invoked (which already needs complete set of bean defining annotations) and also don't add this property to `configurationBuilder`